internal/supervisor/cluster.go

```diff
@@ -73,6 +75,11 @@ func (s *ClusterSupervisor) Do() error {
 		return nil
 	}
 
+	// Sort the clusters by state preference. Relative order is preserved.
+	sort.SliceStable(clusters, func(i, j int) bool {
+		return model.PendingWorkPriority[clusters[i].State] > model.PendingWorkPriority[clusters[j].State]
```

Comment:
Wouldn't this crash if we try to access an unset State key on the `PendingWorkPriority` map? If we try to `PendingWorkPriority[ClusterStateResizeRequested]`, shouldn't this crash? But I see you added tests and those are passing. I'm confused.

Reply:
No. It will return zero value. 

https://go.dev/play/p/koT9sxmFsgP

Reply:
I may have been thinking on something else then. Good good! Thanks!